providers/os/resources/npm.go

```diff
@@ -118,6 +130,34 @@ func collectNpmPackagesInPaths(runtime *plugin.Runtime, fs afero.Fs, paths []str
 	return directPackageList, transitivePackageList, evidenceFiles, nil
 }
 
+// hasLockfile checks for the lock files
+func hasLockfile(runtime *plugin.Runtime, fs afero.Fs, path string) bool {
+	// specific path was provided
+	afs := &afero.Afero{Fs: fs}
+	isDir, err := afs.IsDir(path)
+	if err != nil {
+		return false
+	}
+
+	searchPaths := []string{}
+	if isDir {
+		// check if there is a package-lock.json or package.json file
+		searchPaths = append(searchPaths, filepath.Join(path, "/package-lock.json"))
```

Comment:
we are going to add support for more lock files, therefore we've written the code this way